src/SprykerSdk/Zed/SprykGui/Communication/Controller/AbstractController.php

```diff
@@ -24,25 +22,6 @@ public function initialize(): void
     {
         parent::initialize();
 
-        $this->assertNonProductionEnvironment();
-    }
-
-    /**
-     * @throws \Symfony\Component\HttpKernel\Exception\NotFoundHttpException
-     *
-     * @return void
-     */
-    protected function assertNonProductionEnvironment(): void
-    {
-        $isProductionEnvironment = Environment::isProduction();
-        $isCli = PHP_SAPI === 'cli';
-
-        if (!$isProductionEnvironment || $isCli) {
-            return;
-        }
-
-        throw new NotFoundHttpException(
```

Comment:
What is the reason we are going away from this? The whole ocnfig thing still seems not only overkill but wrong to me.

Reply:
Hi, we are trying to avoid hardcoded (environment names) hidden logic behind environment settings and gives the possibility to change the current behavior of the application. 

Also, it will not be possible to use Environment::methodName().

Reply:
This here is done on purpose as security only measure in the unlikely case someone is using it wrongly. It must never be possible zo change behavior here.
I dont see any reason for this to be changed in the way proposed here.

Reply:
in this case, we have only one case revert it back and use APPLICATION_ENV === 'production' instead of Environment::isProduction()

Reply:
but in this case, it's possible that our environment name could be e.g docker or kubernetus

Reply:
ENV name should always be one of the names development/testing/staging/production, the details shouldnt matter.
Maybe the requirements for docker etc are wrong here?

Reply:
First of all, we do not have ENV type in Spryker. We have only APPLICATION_ENV that was assumed as environment type across many modules. That is a very huge blocker for our customers to use different APPLICATION_ENV for different setups.

Secondly, we've planned to introduce environment type but changed our mind due to:
- Hidden logic. You won't know what stays behind environment type without reverse engineering. It is impossible to write in the documentation because the code is changing every minute, everybody can add one more usage of `isProduction` method.
- Also, we have some logic that cannot be mapped to one environment type. We considered to have multiple types allowed at the same time, to have logic simple without `isProduction() || isStaging()`. But this is hard to maintain during deployment.
- It is better to replace condition based on APPLICATION_ENV to specific configuration parameters, that are obvious and can be mixed in any order in the environment configuration.

Thirdly, in this particular case, we already have a fuse as `spryk` module is in `require-dev` and in a real production environment this code won't be even present in `vendor`, therefore it will not be executed. That is why we proposed to remove this condition, as it has no sense.

Reply:
> Thirdly, in this particular case, we already have a fuse as spryk module is in require-dev and in a real production environment this code won't be even present in vendor, therefore it will not be executed. That is why we proposed to remove this condition, as it has no sense.

That is exactly what I always said: It shouldnt be there if done properly as per specs.
BUT as others pointed out, developers dont always do the right thing, and as such the condition was in here on purpose, as additional security/safety net.
So removing it introduces the issue of being able to enable this again (by accident or on purpose) on production - which shouldnt be possible!

Reply:
If only we would have a trusted source of the `isProduction` knowledge, but the environment name is not a trusted source, unfortunately. We should not decide for clients where and how they run console commands, that is their responsibility. But with this condition, we just make the overhead and making inconvenience for people who do not follow `development/production/testing/staging` pattern.

Reply:
Removing this is fine then for now - BUT we still have lots of other places where similar logic is important and critical to have/keep.
You are currently now yet showing a clean solution to this.

I would expect you to find a way to aggregate possible "OK" values under a method in config named isProduction() to have this way of knowing.
So why not having a whitelist of env configs or names to make up these methods?

Reply:
1. If we place this `production` black/white list on the project level, we give control to the client, and no matter how the client control that, with this list, or by proper deployment scripts, we cannot interfere anymore and guarantee anything related to this.
2. In all other cases, we introduced specific configuration parameters to handle behavior depending on the environment. 
3. In cases similar to this (related to require-dev) the same logic will be applied, everything will rely on `composer` mode.

Thanks for participating. If you are interested in the final epic review, let me know, and I ping you when PRs are ready.

Reply:
I think on core we should default to sane defaults here, project level CAN overwrite if needed, but doesnt have to in all default cases IMO, especially since we need to keep this BC for security reasons for all existing customers that rely on these methods already in production.

Reply:
I would agree with you according to BC. Would you accept the solution to add additional configuration parameter `ENABLE_SPRYK_WEB_INTERFACE` that by default is equal `APPLICATION_ENV !== 'production'`?

Reply:
I still dont think we need this for this case. As we can remove the whole thing as you said, if people correctly deploy.

Reply:
I will wrap this during epic refactoring. Merging the story as it is for now.